spec/models/user_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+describe User do
+  it "validates for an incorrect password length" do
+    user = User.new
+    user.username = Faker::Internet.user_name
+    user.email = Faker::Internet.email
+    user.password = 'abc123'
+
+    expect(user).to_not be_valid
+    expect(user.errors.messages[:password]).to_not be_blank
+  end
+
+  it "validates for an correct password length" do
+    user = User.new
+    user.username = Faker::Internet.user_name
+    user.email = Faker::Internet.email
+    user.password = 'abc123abc'
+
+    expect(user).to be_valid
+    expect(user.errors.messages[:password]).to be_blank
+  end
+
+  it "validates for a non-unique username" do
+    user = FactoryGirl.create(:user)
+    begin
+      user2 = FactoryGirl.create(:user)
+    rescue
+      expect(user2).to be_nil
+    end
```

Comment:
We should really sequence the username in the user factory. I'll make a comment on your factory on how to do this.

```
bob = FactoryGirl.create(:user, username: "bob123")

# creating someone with the same username will raise an error
expect { FactoryGirl.create(:user, username: "bob123") }.to raise_error

# building a user (without saving) will fail validations
expect(FactoryGirl.build(:user, username: "bob123")).to_not be_valid
```